### PR TITLE
[serve] Fix handle JSON ser/de

### PR DESCRIPTION
**Pull request by edoakes**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/pipeline/tests/test_json_serde.py [hunk 1]
```diff
@@ -322,6 +328,44 @@ def test_nested_deployment_node_json_serde(serve_instance):
...
+@pytest.mark.asyncio
+class TestHandleJSON:
+    def test_invalid(self, serve_instance):
+        with pytest.raises(ValueError):
+            serve_handle_from_json_dict({"blah": 123})
+
+    @pytest.mark.parametrize("sync", [False, True])
+    async def test_basic(self, serve_instance, sync):
+        handle = get_handle(sync)
+        await call(handle, "hi") == "hi"
```
Comment (jiaodong) [inline python/ray/serve/pipeline/tests/test_json_serde.py:358]: assert ?

File: python/ray/serve/pipeline/tests/test_json_serde.py [hunk 2]
```diff
@@ -322,6 +328,44 @@ def test_nested_deployment_node_json_serde(serve_instance):
...
+
+        serialized = json.dumps(serve_handle_to_json_dict(handle))
+        # Check we can go through multiple rounds of serde.
+        serialized = json.dumps(json.loads(serialized))
+
+        # Load the handle back from the dict.
+        handle = serve_handle_from_json_dict(json.loads(serialized))
+        await call(handle, "hi") == "hi"
```
Comment (jiaodong) [inline python/ray/serve/pipeline/tests/test_json_serde.py:366]: assert ?

File: python/ray/serve/pipeline/tests/test_json_serde.py [hunk 3]
```diff
@@ -322,6 +328,44 @@ def test_nested_deployment_node_json_serde(serve_instance):
...
+    def test_invalid(self, serve_instance):
+        with pytest.raises(ValueError):
+            serve_handle_from_json_dict({"blah": 123})
+
+    @pytest.mark.parametrize("sync", [False, True])
+    async def test_basic(self, serve_instance, sync):
+        handle = get_handle(sync)
+        await call(handle, "hi") == "hi"
+
+        serialized = json.dumps(serve_handle_to_json_dict(handle))
```
Comment (jiaodong) [inline python/ray/serve/pipeline/tests/test_json_serde.py:360]: can we also add to ensure `json.dumps(handle, cls=DAGNodeEncoder)` works the same ?

**Review by jiaodong — APPROVED** [review]
lgtm, just nits

File: python/ray/serve/pipeline/tests/test_json_serde.py (hunk 1 above)
Comment (edoakes) [reply python/ray/serve/pipeline/tests/test_json_serde.py:358]: ope

File: python/ray/serve/pipeline/tests/test_json_serde.py (hunk 3 above)
Comment (edoakes) [reply python/ray/serve/pipeline/tests/test_json_serde.py:360]: I already have some tests for this in the `serve build` PR, will leave them out here for simplicity

**Comment by simon-mo** [comment]
```


==================== Test output for //python/ray/serve:test_handle:
--
  | Traceback (most recent call last):
  | File "/root/.cache/bazel/_bazel_root/5fe90af4e7d1ed9fcf52f59e39e126f5/execroot/com_github_ray_project_ray/bazel-out/k8-opt/bin/python/ray/serve/test_handle.runfiles/com_github_ray_project_ray/python/ray/serve/tests/test_handle.py", line 10, in <module>
  | from ray.serve.utils import ServeHandleEncoder, serve_handle_object_hook
  | ImportError: cannot import name 'ServeHandleEncoder'


```

**Pull request merged**